### PR TITLE
[DI] Detect circular references with ChildDefinition parent

### PR DESCRIPTION
**Pull request by Seb33300**
| Q             | A
| ------------- | ---
| Branch?       | 3.4 (be careful when merging)
| Bug fix?      | yes
| New feature?  | no
| BC breaks?    | no
| Deprecations? | no
| Tests pass?   | yes
| Fixed tickets | #28312
| License       | MIT
| Doc PR        | 

I will provide a test case if the fix looks good for you :)

### CODE REVIEW

**Comment by Seb33300** [comment]
Oups, didn't see that someone already worked on it: https://github.com/symfony/symfony/pull/28452

**Comment by Seb33300** [comment]
Just pushed a test case since it looks to be the good way to fix this issue regarding to comment on the other issue.

I pushed on master since it is a very similar issue as the previous one I fixed (https://github.com/symfony/symfony/pull/28315) which was merged on master.

File: src/Symfony/Component/DependencyInjection/Tests/Compiler/ResolveChildDefinitionsPassTest.php [hunk 1]
```diff
@@ -396,4 +396,20 @@ protected function process(ContainerBuilder $container)
         $pass = new ResolveChildDefinitionsPass();
         $pass->process($container);
     }
+
+    /**
+     * @expectedException \Symfony\Component\DependencyInjection\Exception\ServiceCircularReferenceException
```
Comment (nicolas-grekas) [inline src/Symfony/Component/DependencyInjection/Tests/Compiler/ResolveChildDefinitionsPassTest.php]: could be worth adding an `@expectedExceptionMessage`: right now, I get: `Service "a": Service "a": Service "a": Service "a": Circular reference detected for service "c", path: "c -> b -> a -> c".`, which might need some tweaks :)

Comment (stof) [reply src/Symfony/Component/DependencyInjection/Tests/Compiler/ResolveChildDefinitionsPassTest.php]: hmm, that's `resolveDefinition` prefixing the exception message. And the multiple prefixing looks really weird though.

Comment (Seb33300) [reply src/Symfony/Component/DependencyInjection/Tests/Compiler/ResolveChildDefinitionsPassTest.php]: Oups, didn't see that. I just pushed a fix.

Comment (nicolas-grekas) [reply src/Symfony/Component/DependencyInjection/Tests/Compiler/ResolveChildDefinitionsPassTest.php]: still worth adding `@expectedExceptionMessage` :)

Comment (Seb33300) [reply src/Symfony/Component/DependencyInjection/Tests/Compiler/ResolveChildDefinitionsPassTest.php]: Maybe `@expectedExceptionMessageRegExp` to test if exception message is not prefixed?

**Review by nicolas-grekas — APPROVED** [review]

**Comment by nicolas-grekas** [comment]
(for 3.4!)

**Review by stof — APPROVED** [review]

**Comment by nicolas-grekas** [comment]
Thank you @Seb33300.

**Pull request merged**